sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -93,13 +70,97 @@ protected override void Initialize(SonarAnalysisContext context)
             (statement as ReturnStatementSyntax)?.ReturnKeyword.ToString() ??
             (statement as ThrowStatementSyntax)?.ThrowKeyword.ToString();
 
-        private static bool IsWithinMethodPropertyOrLambda(SyntaxNode node) =>
-            !node.IsAnyKind(methodOrPropertyDeclarations);
+        private class FooWalker : CSharpSyntaxWalker
+        {
+            private readonly SyntaxNodeAnalysisContext context;

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the 'context' field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1450)